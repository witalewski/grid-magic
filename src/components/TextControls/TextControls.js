import React from 'react';
import { inject, observer } from 'mobx-react';
import { func, string } from 'prop-types';

import styled from '@emotion/styled';

const TextControlsStyled = styled.section`
  margin: 8px;

  .text-controls-content {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 250px;
    height: 80px;
    background: #e6e7e6;
    border: 1px solid #476a6f;
  }

  .text-controls-content > * {
    margin: 8px;
  }

  .section-heading {
    font-weight: normal;
  }
`;

export const TextControls = ({ overlayText, setOverlayText }) => {

  const onChange = ({ target: { value } }) => {
    setOverlayText(value);
  };

  return (
    <TextControlsStyled>
      <h2 className="section-heading">2. Add text</h2>
      <div className="text-controls-content">
        <p>Text:</p>
        <input type="text" value={overlayText} onChange={onChange} />
      </div>
    </TextControlsStyled>
  );
};

TextControls.propTypes = {
  overlayText: string,
  setOverlayText: func.isRequired,
};
TextControls.defaultProps = {
  overlayText: '',
};

export default inject(({ appState }) => ({
  overlayText: appState.overlayText,
  setOverlayText: appState.setOverlayText,
}))(observer(TextControls));
