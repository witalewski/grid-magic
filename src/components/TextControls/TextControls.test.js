import React from 'react';
import { TextControls } from './TextControls';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('TextControls', () => {
  it('displays text controls', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <TextControls setOverlayText={jest.fn()} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
