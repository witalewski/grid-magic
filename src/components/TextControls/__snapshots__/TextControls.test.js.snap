// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextControls displays text controls 1`] = `
<ForwardRef(render)>
  <h2
    className="section-heading"
  >
    2. Add text
  </h2>
  <div
    className="text-controls-content"
  >
    <p>
      Text:
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</ForwardRef(render)>
`;
